spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -184,12 +186,28 @@ public long count(Class<?> entityClass) {
 
 	@Override
 	public <T> Collection<T> findAll(Class<T> entityClass) {
-		return convertEntitiesForRead(
-				this.datastore.run(Query.newEntityQueryBuilder()
-						.setKind(this.datastoreMappingContext
-								.getPersistentEntity(entityClass).kindName())
-						.build()),
-				entityClass);
+		return findAll(entityClass, -1, -1, null);
```

Comment:
Is it ok to put special cases for these trick params (-1) in a public method on the interface? what about `OptionalLong` ?